File: app/src/main/java/com/allrecipes/ui/views/AbstractPresenterView.java
package com.allrecipes.ui.views;

public interface AbstractPresenterView {

    void showToast(int messageResId);

    void showToast(String message);

    void showLoading();

    void hideLoading();

    boolean isFinishing();

    void showConnectivityError();

    void hideConnectivityError();
}

File: app/src/main/java/com/allrecipes/ui/views/AbstractPresenterView.kt
package com.allrecipes.ui.views

interface AbstractPresenterView {

    fun showToast(messageResId: Int)

    fun showToast(message: String)

    fun showLoading()

    fun hideLoading()

    fun isFinishing(): Boolean

    fun handleApiError(throwable: Throwable, predicate : () -> Unit)

    fun hideConnectivityError()
}